actionmailer/lib/action_mailer/base.rb

```diff
@@ -466,30 +466,26 @@ def register_interceptors(*interceptors)
       # Either a class, string or symbol can be passed in as the Observer.
       # If a string or symbol is passed in it will be camelized and constantized.
       def register_observer(observer)
-        delivery_observer = case observer
-          when String, Symbol
-            observer.to_s.camelize.constantize
-          else
-            observer
-          end
-
-        Mail.register_observer(delivery_observer)
+        Mail.register_observer(class_from_value(observer))
       end
 
       # Register an Interceptor which will be called before mail is sent.
       # Either a class, string or symbol can be passed in as the Interceptor.
       # If a string or symbol is passed in it will be camelized and constantized.
       def register_interceptor(interceptor)
-        delivery_interceptor = case interceptor
-          when String, Symbol
-            interceptor.to_s.camelize.constantize
-          else
-            interceptor
-          end
-
-        Mail.register_interceptor(delivery_interceptor)
+        Mail.register_interceptor(class_from_value(interceptor))
       end
 
+      def class_from_value(value)
```

Comment:
Don't think we gain much from having such a broad name. I'd rename to `observer_class_for` :grin: